index.js

```diff
@@ -95,29 +95,56 @@ const baseSchema = {
 }
 
 function mergeJsonSchemas(schema, otherSchema) {
-  const { properties: schemaProperties, required: requiredSchema = [], ...schemaRemainingProperties } = schema
-  const {
-    properties: otherSchemaProperties,
-    required: requiredOtherSchema = [],
-    ...otherSchemaRemainingProperties
-  } = otherSchema
   const mergedSchema = {
-    type: 'object',
-    properties: {
-      ...schemaProperties,
-      ...otherSchemaProperties,
-    },
-    required: [...requiredSchema, ...requiredOtherSchema],
-    allOf: [
-      schemaRemainingProperties,
-      otherSchemaRemainingProperties,
-    ],
     additionalProperties: false,
+    type: 'object',
   }
+
+  Object.keys(schema).forEach(key => {
+    if (Array.isArray(schema[key])) {
+      mergedSchema[key] = [
+        ...mergedSchema[key] ?? [],
+        ...schema[key],
+      ]
+      return true
+    }
+
+    if (typeof schema[key] === 'object') {
+      mergedSchema[key] = {
+        ...mergedSchema[key] ?? {},
+        ...schema[key],
+      }
+      return true
+    }
+
+    mergedSchema[key] = schema[key]
+  })
+
+  Object.keys(otherSchema).forEach(key => {
+    if (Array.isArray(otherSchema[key])) {
+      mergedSchema[key] = [
+        ...mergedSchema[key] ?? [],
+        ...otherSchema[key],
+      ]
+      return true
+    }
+
+    if (typeof otherSchema[key] === 'object') {
+      mergedSchema[key] = {
+        ...mergedSchema[key] ?? {},
+        ...otherSchema[key],
+      }
+      return true
+    }
+
+    mergedSchema[key] = otherSchema[key]
+  })
+
```

Comment:
Correct me if I'm wrong but I think we could just use `[mergeDeepLeft](https://ramdajs.com/docs/#mergeDeepLeft)` from ramda to hide complexity

Reply:
I did a couple of test and it's a bit more complicated since there are arrays that need to be concatenated inside, but this should the the trick for all the cases:
```javascript
const mergedEnv = R.mergeDeepWithKey((key, left, right) => {
  return typeof left === 'object' && typeof right === 'object' ? R.concat(left, right) : right
}, env, otherEnv)
```

Reply:
Fixed